include/votca/xtp/threecenter.h

```diff
@@ -148,9 +149,10 @@ class TCMatrix_gwbse : public TCMatrix {
   std::vector<CudaMatrix> SendDFTMatricesToGPU(
       const Eigen::MatrixXd& dft_orbitals) const;
 
-  void FillBlockCUDA(std::vector<Eigen::MatrixXd>& block,
-                     const std::vector<Eigen::MatrixXd>& symmstorage,
-                     std::vector<CudaMatrix>& cudaMatrices);
+  std::vector<Eigen::MatrixXd> FillBlockCUDA(
+      std::vector<Eigen::MatrixXd>& block,
+      const std::vector<Eigen::MatrixXd>& symmstorage,
+      std::vector<CudaMatrix>& cudaMatrices);
```

Comment:
ahh now I see why you want the vector of all CudaMatrices, I still think it is clearer, if you have two arguments `std::array<CUDAMatrix,2>& dftorbitals, std::array<CUDAMatrix,4>& temporaries` because then they are all clearly distinguishable